client/dashboard/domains/domain-contact-details/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { useQuery } from '@tanstack/react-query';
+import { useNavigate } from '@tanstack/react-router';
+import { __ } from '@wordpress/i18n';
+import { domainWhoisQuery } from '../../app/queries/domain-whois';
+import { domainRoute } from '../../app/routes/domain-routes';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { findRegistrantWhois } from '../../utils/domain-whois';
+import ContactForm from './contact-form';
+import type { DomainContactDetails } from '../../data/domain-whois';
+
+import './style.scss';
+
+export default function DomainContactInfo() {
+	const { domainName } = domainRoute.useParams();
+	const navigate = useNavigate();
+	const { data: whoisData } = useQuery( domainWhoisQuery( domainName ) );
+	const registrantWhoisData = findRegistrantWhois( whoisData );
+
+	const handleSubmit = () => {
+		navigate( { to: '/domains/$domainName', params: { domainName } } );
+	};
+
+	const handleCancel = () => {
+		navigate( { to: '/domains/$domainName', params: { domainName } } );
+	};
+
+	return (
+		<PageLayout size="small" header={ <PageHeader title={ __( 'Contact details' ) } /> }>
+			<div className="domain-contact-info">
```

Comment:
Let's use HStack/VStack instead of custom classes. (I believe we can delete it now since it contains no styles)